pkg/adapter/common/backoff_test.go

```diff
@@ -109,15 +108,16 @@ func TestDuration(t *testing.T) {
 }
 
 func TestRun(t *testing.T) {
-	var counter int
-	start := time.Now()
-
 	testCases := map[string]struct {
 		fn         func(context.Context) (bool, error)
 		waitReturn bool
 		err        error
 	}{
-		"force stop with no errs": {
+		"force stop with no err": {
+			fn: func(ctx context.Context) (bool, error) {
+				// never terminates by itself
+				return true, nil
+			},
```

Comment:
I encountered a race when `fn` was `nil` and the `doneCh` wasn't closed fast enough.